packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol

```diff
@@ -0,0 +1,177 @@
+    // @param _blueprints Addresses of Blueprint contracts
+    // @param _implementations Addresses of implementation contracts
+    // @param _upgradeController Address of the upgrade controller
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        IProtocolVersions _protocolVersions,
+        IProxyAdmin _superchainProxyAdmin,
+        string memory _l1ContractsRelease,
+        Blueprints memory _blueprints,
+        Implementations memory _implementations,
+        address _upgradeController
+    )
+        OPContractsManager(
+            _superchainConfig,
+            _protocolVersions,
+            _superchainProxyAdmin,
+            _l1ContractsRelease,
+            _blueprints,
+            _implementations,
+            _upgradeController
+        )
+    { }
+
+    /// @notice Overrides the l1ContractsRelease function to return "none", as this OPCM
+    /// is not releasing new contracts.
+    function l1ContractsRelease() external pure override returns (string memory) {
+        return "none";
+    }
+
+    function deploy(DeployInput memory _input) external pure override returns (DeployOutput memory) {
+        _input; // Silence warning
+        revert NotImplemented();
+    }
+
+    function upgrade(OpChainConfig[] memory _opChainConfigs) external pure override {
+        _opChainConfigs; // Silence warning
+        revert NotImplemented();
+    }
+
+    function addGameType(AddGameInput[] memory _gameConfigs) public pure override returns (AddGameOutput[] memory) {
+        _gameConfigs; // Silence warning
+        revert NotImplemented();
+    }
+
+    /// @notice Updates the prestate hash for a new game type while keeping all other parameters the same
+    /// @param _prestateUpdateInputs The new prestate hash to use
+    function updatePrestate(OpChainConfig[] memory _prestateUpdateInputs) external {
+        // Loop through each chain and prestate hash
+        for (uint256 i = 0; i < _prestateUpdateInputs.length; i++) {
+            IFaultDisputeGame permissionlessDisputeGame = IFaultDisputeGame(
+                address(
+                    getGameImplementation(
+                        IDisputeGameFactory(_prestateUpdateInputs[i].systemConfigProxy.disputeGameFactory()),
+                        GameTypes.CANNON
+                    )
+                )
+            );
+
+            bool hasFDG = !(address(permissionlessDisputeGame) == address(0));
+
+            AddGameInput[] memory inputs = new AddGameInput[](hasFDG ? 2 : 1);
+            AddGameInput memory pdgInput;
+            AddGameInput memory fdgInput;
+
+            if (Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) == bytes32(0)) {
+                revert PDGPrestateRequired();
+            }
+            // Get the current game implementation to copy parameters from
+            IDisputeGameFactory dgf =
+                IDisputeGameFactory(_prestateUpdateInputs[i].systemConfigProxy.disputeGameFactory());
```

Comment:
we should move this line to the beginning of the `for` loop since that lets deduplicate the earlier `.disputeGameFactory()` call